### ISSUE TITLE
[Bug] Exception: `getCurrentAnkiDroidDirectory`

### ISSUE THREAD

**Issue by david-allison** [ankidroid/Anki-Android#13644]
Almost certainly a `StackOverflowException`, given a runaway stack trace containing:

```
com.ichi2.anki.servicelayer.ScopedStorageService.isLegacyStorage(ScopedStorageService.kt:250)
com.ichi2.anki.StartupStoragePermissionManager$Companion.selectAnkiDroidFolder(InitialActivity.kt:286)
com.ichi2.anki.CollectionHelper$Companion.getDefaultAnkiDroidDirectory(CollectionHelper.kt:465)
com.ichi2.anki.CollectionHelper$Companion.getCurrentAnkiDroidDirectory(CollectionHelper.kt:548)
com.ichi2.anki.servicelayer.ScopedStorageService.isLegacyStorage(ScopedStorageService.kt:250)
```

###### Reproduction Steps

1. AOSP Emulator, API 30. Unknown cause. Likely a build from ce0b0dc9ff6cfad16060e90e5e2f7316a8d478b0

Expected cause: #13630

###### Debug info
Saw this on an emulator logcat after I'd loaded another version of AnkiDroid. None available

###### Research
- [x] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [x] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [x] I have searched for similar existing issues here and on the user forum
- [ ] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

**Comment by david-allison** [comment]
@BrayanDSO could you take a quick look at this when feasible. I noticed this in logcat after I'd installed another build, so apologies for the lack of detail.

The stack trace points to a recent code change. After further review, I don't see a problem (unless there's a race condition). This could potentially be caused by code which was development-only and never made it into `main`, but I find this doubtful given the timing.

**Comment by david-allison** [comment]
Reproduced on an API 32 emulator:

```
04-22 06:23:57.303  7190  7190 E ACRA    : 	at com.ichi2.anki.StartupStoragePermissionManager$Companion.selectAnkiDroidFolder(InitialActivity.kt:286)
04-22 06:23:57.304  7190  7190 E ACRA    : 	at com.ichi2.anki.CollectionHelper$Companion.getDefaultAnkiDroidDirectory(CollectionHelper.kt:465)
04-22 06:23:57.304  7190  7190 E ACRA    : 	at com.ichi2.anki.CollectionHelper$Companion.getCurrentAnkiDroidDirectory(CollectionHelper.kt:548)
04-22 06:23:57.304  7190  7190 E ACRA    : 	at com.ichi2.anki.servicelayer.ScopedStorageService.isLegacyStorage(ScopedStorageService.kt:258)
04-22 06:23:57.304  7190  7190 E ACRA    : 	at com.ichi2.anki.StartupStoragePermissionManager$Companion.selectAnkiDroidFolder(InitialActivity.kt:286)
04-22 06:23:57.304  7190  7190 E ACRA    : 	at com.ichi2.anki.CollectionHelper$Companion.getDefaultAnkiDroidDirectory(CollectionHelper.kt:465)
04-22 06:23:57.304  7190  7190 E ACRA    : 	at com.ichi2.anki.CollectionHelper$Companion.getCurrentAnkiDroidDirectory(CollectionHelper.kt:548)
04-22 06:23:57.304  7190  7190 E ACRA    : 	at com.ichi2.anki.servicelayer.ScopedStorageService.isLegacyStorage(ScopedStorageService.kt:258)
04-22 06:23:57.304  7190  7190 E ACRA    : 	at com.ichi2.anki.StartupStoragePermissionManager$Companion.selectAnkiDroidFolder(InitialActivity.kt:286)
04-22 06:23:57.304  7190  7190 E ACRA    : 	at com.ichi2.anki.StartupStoragePermissionManager$Companion.register(InitialActivity.kt:275)
04-22 06:23:57.304  7190  7190 E ACRA    : 	at com.ichi2.anki.DeckPicker.onCreate(DeckPicker.kt:321)
04-22 06:23:57.304  7190  7190 E ACRA    : 	at android.app.Activity.performCreate(Activity.java:8057)
04-22 06:23:57.304  7190  7190 E ACRA    : 	at android.app.Activity.performCreate(Activity.java:8037)
04-22 06:23:57.304  7190  7190 E ACRA    : 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1341)
04-22 06:23:57.304  7190  7190 E ACRA    : 	at androidx.test.runner.MonitoringInstrumentation.callActivityOnCreate(MonitoringInstrumentation.java:779)
```

**Comment by BrayanDSO** [comment]
@david-allison, do you know the steps to reproduce it?

**Comment by david-allison** [comment]
Not currently, I'll look into it ASAP. Caught it when I was signing off yesterday

**Comment by david-allison** [comment]
* Broken on a86277fdee6998546d4824c2acb98d6ba06e8390
* Fixed on 79db94aafdce2a375597d17647dc70def1af6a75

Fixed by: 6dcf23fd10f2eb5d5e8333da895af8d4725f9397

But: this is because `isLegacyStorage(context, setCollectionPath = false)` is gated on a conditional which now returns false

----

Issue was: https://github.com/ankidroid/Anki-Android/blob/567f131dfdd798bce43c4970c033cb758506731b/AnkiDroid/src/main/java/com/ichi2/anki/CollectionHelper.kt#L546-L551 , recursive unconditional call on `getDefaultAnkiDroidDirectory`
